minbpe/basic.py

```
@@ -11,17 +11,79 @@
 
+class BasicTokenizer(Tokenizer):
     def __init__(self):
-        raise NotImplementedError()
+        super().__init__()
+
+    @staticmethod
+    def _simple_num_merges(vocab_size: int) -> int:
+        return vocab_size - MIN_VOCAB_SIZE
+
+    def train(
+        self, text: str, vocab_size: int, num_merges: int = None, verbose: bool = False
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)